### PR TITLE
[BE][Easy] fix ruff rule needless-bool (SIM103)

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130206

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @LucasLLC @MeetVadakkanchery @mhorowitz @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @tianyu-l

### CODE REVIEW

**Comment by ezyang** [comment]
I am also suspicious about this one. If E: return True; return False can make sense if you have exploded a disjunction into multiple if statements with short circuiting returns. It's unnatural to fold the last short circuit into an expression in this case.

**Comment by XuehaiPan** [comment]
> If E: return True; return False can make sense if you have exploded a disjunction into multiple if statements with short circuiting returns. It's unnatural to fold the last short circuit into an expression in this case.

Reverted related changes.

File: test/jit/test_enum.py [hunk 1]
```diff
@@ -175,10 +175,7 @@ class Color(Enum):
         make_global(Color)
 
         def enum_const(x: Color) -> bool:
-            if x == Color.PURPLE:
-                return True
-            else:
-                return False
+            return x == Color.PURPLE
```
Comment (malfet) [inline test/jit/test_enum.py:178]: I would not touch jit tests, because may be some of those is validating that frontend parser can deal with those constructions

File: test/test_jit.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def doAutodiffCheck(testname):
     # these tests are disabled because BailOut nodes
     # inserted by ProfilingExecutor interfere with
     # subgraph slicing of Differentiable Graphs
-    test_exceptions = [
+    test_exceptions = (
```
Comment (malfet) [inline test/test_jit.py:170]: Not sure why you need to replace list with tuple here, but sure, why not

File: torch/_inductor/fx_passes/quantization.py [hunk 3]
```diff
@@ -607,13 +607,7 @@ def fn(match):
...
         )
         if (
-            len(
-                _get_remaining_users(
-                    extra_input_of_pattern,
-                    compute_node,
-                )
-            )
-            > 1
+            len(_get_remaining_users(extra_input_of_pattern, compute_node)) > 1
```
Comment (malfet) [inline torch/_inductor/fx_passes/quantization.py:610]: This change is unrelated to the rest of the PR, isn't it?

File: scripts/compile_tests/common.py [hunk 4]
```diff
@@ -43,10 +43,7 @@ def find(testcase, condition):
 
 def skipped_test(testcase):
     def condition(children):
-        tags = [child.tag for child in children]
-        if "skipped" in tags:
-            return True
-        return False
+        return "skipped" in {child.tag for child in children}
```
Comment (malfet) [inline scripts/compile_tests/common.py:46]: Replacing list with set is a bit unrelated, though reasonable
```suggestion
        return "skipped" in {child.tag for child in children}
```

**Review by malfet — APPROVED** [review]
LGTM, though please avoid making unrelated changes (one formatting change, few replacing lists with sets or tuples) and consider keeping jit tests as is, as they might be testing some parser features
